### ISSUE TITLE
When will json global variable be back

### ISSUE THREAD

**Issue by kykungz** [callumlocke/json-formatter#231]
The README.md stated that:

> - ~Parsed JSON is exported as a global variable, json, so you can inspect it in the console~ Disabled for now, due to difficulties getting it working with Manifest v3 upgrade.

But is there an estimated date that it will be back?

**Comment by ChrisBAshton** [comment]
I've found a workaround for now - paste the following into your console:

```js
json = JSON.parse(document.getElementById("jsonFormatterRaw").querySelector("pre").innerText)
```

**Comment by rishiraj88** [comment]
Thanks, @ChrisBAshton !

**Comment by c4ldas** [comment]
The workaround provided by @ChrisBAshton works very well. I suggest the rep owner add this information on the README.md file as a suggestion while the global variation option is not available again. I'm glad I found this information here! :)

**Comment by jespertheend** [comment]
One potential way of doing this would be to add a `setInterval` with a `debugger;` statement.
That way DevTools switches to the context of the extension the moment it is opened:
```js
setInterval(() => {
	debugger;
}, 500);
```

Downside of this is that the page is no longer interactive. But you could measure how long it takes to 'execute' the debugger statement using `perfomance.now()` and stop the interval the moment it takes more than a couple of milliseconds.
That way users only have to resume the debugger only once:

```js
const interval = setInterval(() => {
  const then = performance.now()
  debugger
  if (performance.now() - then > 100) {
    clearInterval(interval)
  }
}, 500)
```

**Comment by jespertheend** [comment]
Alternatively it seems like [`chrome.scripting.executeScript()`](https://developer.chrome.com/docs/extensions/reference/scripting/#method-executeScript) might make it possible to inject a static script included with the extension. But it requires a background script, so I didn't actually try this. I'm not sure whether injected scripts like these have access to the global scope of the page.

I did try the debugger statement though and that seems to work.

**Comment by lionel-rowe** [comment]
> I've found a workaround for now - paste the following into your console:
> 
> ```js
> json = JSON.parse(document.getElementById("jsonFormatterRaw").querySelector("pre").innerText)
> ```

Tampermonkey version that will automatically run on any `.json` URL:

```js
// ==UserScript==
// @name         JSON formatter `json` global
// @namespace    https://github.com/lionel-rowe/
// @version      0.1
// @description  try to take over the world!
// @author       https://github.com/lionel-rowe/
// @match        *://*/*.json
// @icon         https://www.google.com/s2/favicons?sz=64&domain=json.org
// @grant        none
// ==/UserScript==

const el = document.querySelector('#jsonFormatterRaw pre');
if (el) window.json ??= JSON.parse(el.textContent);
```

**Comment by jespertheend** [comment]
How did tampermonkey manage to work around the manifest changes? Are they still using manifest v2?

**Comment by lionel-rowe** [comment]
> How did tampermonkey manage to work around the manifest changes? Are they still using manifest v2?

Given [Tampermonkey #644 [Chrome] Manifest V3: examine the effects](https://github.com/Tampermonkey/tampermonkey/issues/644) is still open, I guess so. It looks like [Google has delayed the retirement of v2](https://groups.google.com/u/0/a/chromium.org/g/chromium-extensions/c/zQ77HkGmK9E?pli=1), though it doesn't seem like there's currently a concrete updated timeline.

**Comment by kykungz** [comment]
While we are stuck with the new Chrome manifest v3 limitation, perhaps we could push a quick and dirty solution to ease things a little, so that users won't need to manually copy-paste the script or create their own bookmarklet?

For example, adding a button that injects the JSON to `window.json`:
![Screenshot 2566-04-25 at 19 38 23](https://user-images.githubusercontent.com/9304909/234280507-32d0abae-6da9-40e7-9037-514baf2b1e97.png)

**Comment by jespertheend** [comment]
@kykungz I don't think that would work. After all, if you can inject a button, why not just inject a `<script>` tag?

**Comment by kykungz** [comment]
> @kykungz I don't think that would work. After all, if you can inject a button, why not just inject a `<script>` tag?

Good point, I've never develop an extension before, so I'm not really sure how it works haha.

For what it's worth,

I tried attaching a click listener to a button, and yes, it does not work (`window` variable seems unaffected).

So I tried again by injecting the script right into the HTML attribute like this:
```js
const buttonInject = document.createElement('button')
buttonInject.innerText = 'Inject window.json'
buttonInject.setAttribute('onclick', `window.json = ${JSON.stringify(parsedJsonValue)}`)
optionBar.appendChild(buttonInject)
```

and it seems to work.

Does this help?

**Comment by JohnRDOrazio** [comment]
I just found that selecting the contents of `pre` doesn't work for large JSON, it gets truncated at a certain point with `...`. So `const json = JSON.parse(document.getElementById("jsonFormatterRaw").querySelector("pre").innerText)` doesn't always work.
